src/frontend/src/lib/types/coingecko.ts

```diff
@@ -53,3 +53,12 @@ export interface CoingeckoSimplePrice {
 }
 
 export type CoingeckoSimplePriceResponse = Record<CoingeckoCoinsId | string, CoingeckoSimplePrice>;
+
+export interface CoingeckoSimpleTokenPrice extends CoingeckoSimplePrice {
+	usd_market_cap: number;
+}
+
+export type CoingeckoSimpleTokenPriceResponse = Record<
```

Comment:
Just an idea. What do you think?

```
export type CoingeckoResponse<T> = Record<CoingeckoCoinsId | string, T>;
```

which you can use in both `CoingeckoSimplePriceResponse` and `CoingeckoSimpleTokenPriceResponse`

```
export type CoingeckoSimpleTokenPriceResponse = CoingeckoResponse<CoingeckoSimpleTokenPrice>
```

that way you can declare `fetchCoingecko` as returning a type extending response

```
const fetchCoingecko = async <T extends >({
  ...
  }): Promise<CoingeckoResponse<T> | null> => {
```

and modify caller to pass the inner type for T

```
export const simpleTokenPrice = async ({
	id,
	contract_addresses,
	...rest
}: CoingeckoSimpleTokenPriceParams): Promise<CoingeckoSimpleTokenPriceResponse | null> =>
	fetchCoingecko<CoingeckoSimpleTokenPrice>({
```

The advantage is that `fetchCoingecko` returns become a bit more descriptive.

Reply:
agreed! one doubt: why would `fetchCoingecko` have a type extending? can it be the same if:

```typescript
const fetchCoingecko = async <T>({
	endpointPath,
	queryParams
}: {
	endpointPath: string;
	queryParams: string;
}): Promise<CoingeckoResponse<T> | null> => {
...
```

Reply:
ah! i saw your next comment!